### PR TITLE
OLP 284

### PR DESCRIPTION
**Pull request by harshad1ledger**

### CODE REVIEW

File: data/balance/coin.go [hunk 1]
```diff
@@ -114,24 +116,27 @@ func (coin Coin) Equals(value Coin) bool {
...
 		debug.PrintStack()
 		logger.Fatal("Invalid Coin", coin)
 	}
 
 	if coin.Currency.Name != value.Currency.Name {
 		//logger.Error("Mismatching currencies", "coin", coin, "value", value)
-		logger.Fatal("Mismatching currencies", coin, value)
-		return coin
+		logger.Error("Mismatching currencies", coin, value)
+		return coin, errors.New("")
```
Comment (alexjlan) [inline data/balance/coin.go]: error info empty

File: data/balance/coin.go [hunk 2]
```diff
@@ -114,24 +116,27 @@ func (coin Coin) Equals(value Coin) bool {
 }
 
 // Minus two coins
-func (coin Coin) Minus(value Coin) Coin {
+func (coin Coin) Minus(value Coin) (Coin, error) {
```
Comment (alexjlan) [inline data/balance/coin.go]: if you change the Minus return type, probably also change the Plus, and so on

File: data/balance/coin.go (hunk 1 above)
Comment (harshad1ledger) [reply data/balance/coin.go]: fixed this

File: data/balance/coin.go (hunk 2 above)
Comment (harshad1ledger) [reply data/balance/coin.go]: fixed this

**Pull request merged**